src/Translation/CustomTranslationAchievementListener.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Translation;
+
+use App\Project\ProgramManager;
+use App\User\Achievements\AchievementManager;
+use Exception;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\TerminateEvent;
+
+class CustomTranslationAchievementListener
+{
+  private AchievementManager $achievement_manager;
+  private ProgramManager $program_manager;
+
+  public function __construct(AchievementManager $achievement_manager, ProgramManager $program_manager)
+  {
+    $this->achievement_manager = $achievement_manager;
+    $this->program_manager = $program_manager;
+  }
+
+  public function onTerminateEvent(TerminateEvent $event): void
+  {
+    $status_code = $event->getResponse()->getStatusCode();
+    if (200 !== $status_code) {
+      return;
+    }
+
+    $request = $event->getRequest();
+
+    if ($this->isCreateCustomTranslationAction($request)) {
+      $project_id = $request->attributes->get('id');
+      if (null === $project_id) {
+        return;
+      }
+
+      $project = $this->program_manager->find($project_id);
+
+      if (null === $project) {
+        return;
+      }
+
+      try {
+        $this->achievement_manager->unlockAchievementCustomTranslation($project->getUser());
```

Comment:
getUser could return null which then would result in a type error @runtime. Either directly catch the throwable or check first if the return value of getUser is null